rllib/evaluation/collectors/agent_collector.py

```diff
@@ -448,14 +470,20 @@ def build_for_training(
 
                 # count computes the number of time steps that we need to consider.
                 # if batch_repeat_value = 1, this number should be the length of
-                # episode so far, which is len(buffer) - shift_before.
+                # episode so far, which is len(buffer) - shift_before (-1 if this
+                # value was gained during inference. This is because we keep a dummy
+                # value at the last position of the buffer that makes it one longer).
                 count = int(
                     math.ceil(
-                        (len(d) - self.shift_before) / view_req.batch_repeat_value
+                        (
+                            len(d)
+                            - int(data_col in self.gained_from_inference)
+                            - self.shift_before
+                        )
+                        / view_req.batch_repeat_value
```

Comment:
There is going to be a follow up that makes it clearer what is happening here in the docs.
Most importantly: During inference some values for the current timestep are missing when building the batch. We account for this with a dummy value that is disregarded when building for training (where all values are present).